### PR TITLE
#5704 Enable google contact search in firefox

### LINKED ISSUE #5704: Enable Google Contact Search -> Your platform is not supported
image

> Your platform is not supported: browser does not support extension windows

nothing shows in console

`Version 8.5.4`

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR fixed enable google contact search issue in firefox

close #5704 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/browser/env.ts [hunk 1]
```diff
@@ -41,7 +41,12 @@ export class Env {
   }
 
   // Check if the current context is a Service Worker
-  public static isBackgroundPage() {
+  public static async isBackgroundPage() {
+    // In firefox, window&document is not null even in background page
+    if (typeof browser !== 'undefined' && typeof browser.runtime?.getBackgroundPage !== 'undefined') {
```
Comment (sosnovsky) [inline extension/js/common/browser/env.ts:46]: this check probably can be simplified to just `if (browser?.runtime?.getBackgroundPage !== undefined)`

**Review by sosnovsky — APPROVED** [review]
Works well now, thanks!

File: extension/js/common/browser/env.ts (hunk 1 above)
Comment (ioanmo226) [reply extension/js/common/browser/env.ts:46]: It would throw `Uncaught ReferenceError: browser is not defined` in chrome.

Comment (sosnovsky) [reply extension/js/common/browser/env.ts:46]: Ah, I see, then all good 👍

**Pull request merged**